postprocessing/resstockpostproc/process_metadata.py

```diff
@@ -105,87 +126,90 @@ def get_transformed_cols(df: pl.LazyFrame, col_maps: Sequence[dict]) -> pl.LazyF
     transformed_cols = []
     all_cols = df.collect_schema().names()
     for col_map in col_maps:
-        if col_map['column_type'] not in ['Input', 'Output']:
+        if col_map["column_type"] not in ["Input", "Output"]:
             continue
-        assert col_map['column_name'] is not None, "ResStock column name must be provided for Input or Output columns"
-        if col_map['column_name'] not in all_cols:
+        assert (
+            col_map["column_name"] is not None
+        ), "ResStock column name must be provided for Input or Output columns"
+        if col_map["column_name"] not in all_cols:
             continue
-        if col_map['conversion_factor']:
-            new_col = (pl.col(col_map['column_name']).cast(pl.Float64) * float(col_map['conversion_factor'])).alias(col_map['published_name'])
+        if col_map["conversion_factor"]:
+            new_col = (
+                pl.col(col_map["column_name"]).cast(pl.Float64)
+                * float(col_map["conversion_factor"])
+            ).alias(col_map["published_name"])
         else:
-            new_col = pl.col(col_map['column_name']).alias(col_map['published_name'])
+            new_col = pl.col(col_map["column_name"]).alias(col_map["published_name"])
         transformed_cols.append(new_col)
-    
-    upgrade_cols = [col for col in all_cols if col.startswith("upgrade_costs.") and col.endswith("_name")]
+
+    upgrade_cols = [
+        col
+        for col in all_cols
+        if col.startswith("upgrade_costs.") and col.endswith("_name")
+    ]
     transformed_cols.extend([pl.col(col).alias(col) for col in upgrade_cols])
     return df.select(transformed_cols)
 
+
 def add_income_and_burden(df: pl.LazyFrame) -> pl.LazyFrame:
+    df = assign_representative_income(df)
+    income_col = "in.representative_income"
+
     new_cols = []
-    
-    # Handle income parsing with better error handling
-    income_expr = pl.when(pl.col("in.income").is_in(["Not Available", ""]))
-    income_expr = income_expr.then(None)
-    
-    # Handle range case like "10000-20000"
-    income_expr = income_expr.when(pl.col("in.income").str.contains("-"))
-    income_expr = income_expr.then(
-        (pl.col("in.income").str.extract(r"(\d+)-(\d+)", 1).cast(pl.Float64, strict=False) +
-         pl.col("in.income").str.extract(r"(\d+)-(\d+)", 2).cast(pl.Float64, strict=False)) / 2
-    )
-    
-    # Handle plus case like "200000+"
-    income_expr = income_expr.when(pl.col("in.income").str.contains("\+"))
-    income_expr = income_expr.then(
-        pl.col("in.income").str.extract(r"(\d+)\+", 1).cast(pl.Float64, strict=False)
+    # Reassign negative or zero dollar income to 1 dollar
```

Comment:
There's a bunch of changes to this file b/c I formatted it with black.

The core change is here. In addition to the income mapper, I reassign 0 and negative income to 1 so we can approximate energy burden. @rajeee